packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -207,4 +253,37 @@ private Map<String, RouteParamType> getRouteParameters(
         });
         return routeParameters;
     }
+
+    boolean isClientMenuUsed(VaadinService vaadinService) {
+        try {
+            String fileRoutesJson = readFileRoutesJsonFile(
+                    ApplicationConfiguration.get(vaadinService.getContext()));
+            var json = mapper.readTree(fileRoutesJson);
+            // as the client registry has no information about layouts, the JSON
+            // is parsed again to search for a root node with children, which
+            // means that we have a root layout.
+            return json.has(0) && json.get(0).has("children")
```

Comment:
Can we be confident, that the layout object is always on the first position in the array?

Reply:
Good point. By definition, regarding the automatic menu, we only consider the existence of a `@layout.tsx` (or `.jsx`) directly under the `Frontend/views/` directory (though, `@layout.tsx` can exist in deeper levels, as well). If such a layout exists, it will be the parent of all the other route objects.  Then, shape of the objects in `file-routes.json` is as:
```json
[
    {
        "route": "",
        "params": {},
        "title": "Main Layout",
        "children": [
             {
                  ...
             },
        ]
    }
]
```
or
```json
[
    {
        "route": "",
        "params": {},
        "title": "Main Layout",
        "children": []
    }
]
```
@platosha can say wether there would any possibility about having more than one root-level objects in the array? Or is it defined as `[]` just to support the empty an array (due to other technical considerations)?

Reply:
The position of the main layout as the first item in the array isn't guaranteed anymore. One practical case is prepending a `/login` route in the root level before the main layout.

Reply:
Could we then change the implementation to the following: iterate through the objects in the root array and look for the one that has non-empty children ? If it's found, then we have a layout.

Reply:
See https://github.com/vaadin/hilla/pull/2338#discussion_r1576234329